gpkit/nomials/nomial_math.py

```diff
@@ -715,8 +718,10 @@ def __init__(self, left, right):
     def as_posyslt1(self):
         "Returns the posys <= 1 representation of this constraint."
         # todo deal with substitutions
-        raise TypeError("SignomialEquality could not simplify to"
-                        " a PosynomialInequality")
+        raise InvalidGPConstraint("SignomialEquality could not simplify"
+                                  " to a PosynomialInequality; try forming"
+                                  " your program as SignomialProgram or"
+                                  " calling .localsolve().")
```

Comment:
how about `try calling .localsolve instead of solve to form your model as a SignomialProgram`